### DISCUSSION TITLE
Thermostats

### DISCUSSION THREAD

**Discussion by mattxjones** [milo526/homebridge-tuya-web#136] [Q&A] [answered]
Hi,

didn’t want to raise an issue as it seemed a little formal - do I understand correctly from the description of the plug in that you don’t current support pulling thermostat devices from smart life to homebridge?

**Comment by milo526** [comment, marked as answer]
Hey;

The "latest" release does indeed not support this.

There is however a newer release (0.5.2, latest is 0.4.8) that does start to add support for them.

See also #63

**Reply by mattxjones** [reply]
Thanks Milo, hugely appreciated!

**Reply by sgruzlewski** [reply]
would love to see this supported. any way to donate or something?

**Reply by milo526** [reply]
Hey @sgruzlewski,

As of 4 days ago, they are supported!
Please make sure that you have version 0.7.0 of the plugin installed.

A donation is always welcome but never required, if you wish to donate there are a few links on the readme page.
Most people prefer to donate via [my PayPal page](https://paypal.me/MCesar)

**Reply by sgruzlewski** [reply]
Hi, I'm using 0.7. It let's me set the temperature but when I close and reopen apple home app it shows "set to (min temp)" where min tem is the value I set in config as minimal temperature. All is fine if I leave the app running in the background but for example I set temp to 20C, it updates and shows correctly, when i kill the home app and reopen it - it shows me "set to 10C"

**Reply by milo526** [reply]
@sgruzlewski 

Did you manage to find the configuration for these devices? 
![Screenshot of the homebridge-tuya-web plugin settings page](https://user-images.githubusercontent.com/5261909/103180005-e7b83900-4891-11eb-9705-4f2bebf8d15e.png)

Playing with the min-, max- temperatures as well as the temperature factors should probably help you!

It sounds an awful lot like either your minimal temperature is not set low enough (this would be the case if your actual temperature is below 10 degrees) or that tuya messes up how it reports your temperature.

Would you mind following the steps under [Supported Device Types](https://github.com/milo526/homebridge-tuya-web#supported-device-types) in the documentation and post the output? That way I can probably see what is going on 😉 Or at least we can work from there!

**Reply by sgruzlewski** [reply]
So my devices extracted with that python script

I have my temperature factor set to 0.1, min is 5, max is 30.

If I set my target temp factor to 0.1 sometimes I get that "set temp to 5"
If I set my target temp factor to 1 sometimes I get "set temp to 30"

```json
{
   "header":{
      "code":"SUCCESS",
      "payloadVersion":1
   },
   "payload":{
      "devices":[
         {
            "data":{
               "current_temperature":205,
               "max_temper":700,
               "min_temper":10,
               "online":True,
               "temperature":200
            },
            "dev_type":"climate",
            "ha_type":"climate",
            "icon":"https://images.tuyaeu.com/smart/program_category_icon/wk.png",
            "id":"bf8669761d0e2c1e7df6mc",
            "name":"Kuchnia"
         },
         {
            "data":{
               "current_temperature":215,
               "max_temper":700,
               "min_temper":10,
               "online":True,
               "temperature":220
            },
            "dev_type":"climate",
            "ha_type":"climate",
            "icon":"https://images.tuyaeu.com/smart/program_category_icon/wk.png",
            "id":"bfe00ef9867413b1e2zxnd",
            "name":"Sypialnia"
         },
         {
            "data":{
               "current_temperature":225,
               "max_temper":700,
               "min_temper":10,
               "online":True,
               "temperature":220
            },
            "dev_type":"climate",
            "ha_type":"climate",
            "icon":"https://images.tuyaeu.com/smart/program_category_icon/wk.png",
            "id":"bf559fec6f868b67cavp9c",
            "name":"Salon"
         },
         {
            "data":{
               "current_temperature":225,
               "max_temper":700,
               "min_temper":10,
               "online":True,
               "temperature":220
            },
            "dev_type":"climate",
            "ha_type":"climate",
            "icon":"https://images.tuyaeu.com/smart/program_category_icon/wk.png",
            "id":"bfec60472454240230c5ec",
            "name":"Łazienka"
         },
         {
            "data":{
               "current_temperature":215,
               "max_temper":700,
               "min_temper":10,
               "online":True,
               "temperature":220
            },
            "dev_type":"climate",
            "ha_type":"climate",
            "icon":"https://images.tuyaeu.com/smart/program_category_icon/wk.png",
            "id":"bf8ee6cd43bc9d5b653tkw",
            "name":"Pieczara"
         }
      ],
      "scenes":[
         
      ]
   }
}
```

My log when the issue occurred. In this case I had target temp set to 0.1 I see that it updated to 2.3000000000000003 but if i would change this to 1 or even 10 the problem happens in the other way around - set temp to (max_temp) instead of (min_temp).

I even tried to change app from smartlife to tuya but no effect - behaves in the same way.

> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Requesting device state
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Creating new debounced promise
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Triggering debouncedDeviceStateRequest
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Requesting device state
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Triggering debouncedDeviceStateRequest
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Requesting device state
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Triggering debouncedDeviceStateRequest
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Requesting device state
> [12/28/2020, 12:01:00 AM] [TuyaWebPlatform] [Salon] - Triggering debouncedDeviceStateRequest
> [12/28/2020, 12:01:00 AM] [Broadlink RM] pilotYamahaOptical getSwitchState: false
> [12/28/2020, 12:01:00 AM] [Yamaha RN-303D] HTTP GetStatus result:On
> [12/28/2020, 12:01:01 AM] [Yamaha RN-303D] HTTP GetStatus result:NaN
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] - Unsetting debouncedDeviceStateRequestPromise
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] - Renewing cache due to RateLimitError
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.TargetTemperature - [GET] 23
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.TargetTemperature - [UPDATE] 2.3000000000000003
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.CurrentHeatingCoolingState - [GET] [object Object]
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.CurrentHeatingCoolingState - [UPDATE] HEAT
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.TargetHeatingCoolingState - [GET] [object Object]
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.TargetHeatingCoolingState - [UPDATE] AUTO
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.CurrentTemperature - [GET] 225
> [12/28/2020, 12:01:01 AM] [TuyaWebPlatform] [Salon] Characteristic.CurrentTemperature - [UPDATE] 22.5

**Reply by milo526** [reply]
Hmm that makes no sense 😝 

```json
{
   "current_temperature":225,
   "max_temper":700,
   "min_temper":10,
   "online":True,
   "temperature":220
}
```

This data is what Tuya sends to the plugin.
The `current_temperature` that is send in the data should be read by the plugin as the current temperature (that seems to work).
The `temperature` that is send in the data should be read by the plugin as the target temperature. This is weirdly parsed as 23!? after which the factor is applied and it is reporting to Homekit that the current temperature is `2.3`.

Could you please [open an issue](https://github.com/milo526/homebridge-tuya-web/issues/new?assignees=&labels=bug&template=bug_report.md&title=Target%20temperature%20mis-reporting%20with%20configured%20target-temperature-factor) for this?
It warrants some better tracking then we can do in the discussions tab.
